### DISCUSSION TITLE
Behaviour of --no-remote-delete when deleting local files

### DISCUSSION THREAD

**Discussion by SylvainLaugie** [abraunegg/onedrive#1381] [Q&A] [answered]
Hi !

I want to use onedrive as an uploader only to MS OneDrive, to send local backups. As you saw, I started with skilion's onedrive and added uploadOnly and keepRemote features. Then, on your advice, I switched to your onedrive. I noticed the option "--remove-source-files" so I also started using it.

In the end, I use the following command command to upload (I'll remove --verbose later):
`onedrive --synchronize --verbose --upload-only --no-remote-delete --remove-source-files`

It works well, except for one thing : local files are deleted (as expected), **but they stay in the database**. Because of that, each new run of the command will still mention those deleted files:

```
Processing MyFolder/MySubfolder/Mybackup-1.txt
The file has been deleted locally
Skipping remote file delete as --upload-only & --no-remote-delete configured
```

I don't know if this is expected behaviour (I feel like it is), so I wanted to discuss this with you. I don't think that we should keep the locally deleted files in the database; if I deleted them that's because I want to forget about them locally.

What do think about this ?

Thanks

**Comment by abraunegg** [comment, marked as answer]
@SylvainLaugie 
> It works well, except for one thing : local files are deleted (as expected), but they stay in the database. Because of that, each new run of the command will still mention those deleted files

That sounds like a bug - please submit a bug report using the following process: https://github.com/abraunegg/onedrive#reporting-an-issue-or-bug

**Reply by SylvainLaugie** [reply]
Opened as #1382
